src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/tools/monitoring/outputMonitor.ts

```diff
@@ -260,6 +261,11 @@ export class OutputMonitor extends Disposable implements IOutputMonitor {
 			currentInterval = Math.min(currentInterval * 2, maxInterval);
 
 			buffer = execution.getOutput();
```

Comment:
I thought we changed this to use onData instead? Constructing lines is relatively expensive and is scales badly as the output grows.

Reply:
looking into using `waitForIdle` now